cosmwasm/contracts/crosschain-registry/src/execute.rs

```diff
@@ -1278,18 +1324,121 @@ mod tests {
             operations: vec![ChainToBech32PrefixInput {
                 operation: FullOperation::Set,
                 chain_name: "OSMOSIS".to_string(),
-                prefix: Some("OSMO".to_string()),
+                prefix: "OSMO".to_string(),
                 new_prefix: None,
             }],
         };
         let info = mock_info(CREATOR_ADDRESS, &[]);
-        contract::execute(deps.as_mut(), mock_env(), info, msg).unwrap();
+        contract::execute(deps.as_mut(), mock_env(), info.clone(), msg).unwrap();
+
+        assert_eq!(
+            CHAIN_TO_BECH32_PREFIX_MAP
+                .load(&deps.storage, "osmosis")
+                .unwrap(),
+            ("osmo", true).into()
+        );
+        assert_eq!(
+            CHAIN_TO_BECH32_PREFIX_REVERSE_MAP
+                .load(&deps.storage, "osmo")
+                .unwrap(),
+            vec!["osmosis"]
+        );
+
+        // Set another chain with the same prefix
+        let msg = ExecuteMsg::ModifyBech32Prefixes {
+            operations: vec![ChainToBech32PrefixInput {
+                operation: FullOperation::Set,
+                chain_name: "ISMISIS".to_string(),
+                prefix: "OSMO".to_string(),
+                new_prefix: None,
+            }],
+        };
+        contract::execute(deps.as_mut(), mock_env(), info.clone(), msg).unwrap();
+
+        assert_eq!(
+            CHAIN_TO_BECH32_PREFIX_MAP
+                .load(&deps.storage, "ismisis")
+                .unwrap(),
+            ("osmo", true).into()
+        );
+        assert_eq!(
+            CHAIN_TO_BECH32_PREFIX_REVERSE_MAP
+                .load(&deps.storage, "osmo")
+                .unwrap(),
+            vec!["osmosis", "ismisis"]
+        );
 
+        // Set another chain with the same prefix
+        let msg = ExecuteMsg::ModifyBech32Prefixes {
+            operations: vec![ChainToBech32PrefixInput {
+                operation: FullOperation::Disable,
+                chain_name: "OSMOSIS".to_string(),
+                prefix: "OSMO".to_string(),
+                new_prefix: None,
+            }],
+        };
+        contract::execute(deps.as_mut(), mock_env(), info.clone(), msg).unwrap();
+        assert_eq!(
+            CHAIN_TO_BECH32_PREFIX_MAP
+                .load(&deps.storage, "osmosis")
+                .unwrap(),
+            ("osmo", false).into()
```

Comment:
Why is this set to false here?

Reply:
Because the operation above is disable: https://github.com/osmosis-labs/osmosis/blob/edad03916cfab32aecabc6dfa6f1b34924b64cb0/cosmwasm/contracts/crosschain-registry/src/execute.rs#L1374